src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -378,31 +381,39 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         schema = [Attribute('"$1"', VariantType())]
         read_file_transformations = None
         schema_to_cast = None
+        create_tmp_file_format_query: Optional[str] = None
+        drop_tmp_file_format_if_exists_query: Optional[str] = None
         if self._infer_schema:
             temp_file_format_name = (
                 self._session.get_fully_qualified_current_schema()
                 + "."
                 + random_name_for_temp_object(TempObjectType.FILE_FORMAT)
             )
-            create_file_format_query = create_file_format_statement(
-                temp_file_format_name,
-                format,
-                format_type_options,
-                temp=True,
-                if_not_exist=True,
-            )
-            drop_file_format_if_exists_query = drop_file_format_if_exists_statement(
-                temp_file_format_name
+            use_temp_file_format = "FORMAT_NAME" not in self._cur_options
+            file_format_name = self._cur_options.get(
+                "FORMAT_NAME", temp_file_format_name
             )
-            infer_schema_query = infer_schema_statement(path, temp_file_format_name)
-            try:
-                self._session._conn.run_query(
-                    create_file_format_query, is_ddl_on_temp_object=True
+            if use_temp_file_format is not None:
```

Comment:
`use_temp_file_format` is a `bool` so probably we could just do `if use_temp_file_format`?

Reply:
Thanks for catching this, at first it was not a bool :)